desktop/ui/src/main/java/org/datacleaner/windows/DataCloudLogInWindow.java

```diff
@@ -0,0 +1,294 @@
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.windows;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+
+import javax.inject.Inject;
+import javax.swing.*;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.configuration.DataCleanerConfiguration;
+import org.datacleaner.configuration.RemoteServerData;
+import org.datacleaner.descriptors.RemoteDescriptorProvider;
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.restclient.ComponentRESTClient;
+import org.datacleaner.user.UserPreferences;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImageManager;
+import org.datacleaner.util.RemoteServersConfigUtils;
+import org.datacleaner.util.RemoteServersUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.widgets.DCHtmlBox;
+import org.jdesktop.swingx.JXLabel;
+import org.jdesktop.swingx.JXTextField;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataCloudLogInWindow extends AbstractDialog {
+    private static final Logger logger = LoggerFactory.getLogger(DataCloudLogInWindow.class);
+    public static final String SHOW_DATACLOUD_DIALOG_USER_PREFERENCE = "show.datacloud.dialog";
+
+    final private DataCleanerConfiguration _configuration;
+    final private UserPreferences _userPreferences;
+    final private JComponent _contentPanel;
+    private JEditorPane invalidCredentialsLabel;
+    private JXTextField usernameTextField;
+    private JPasswordField passwordTextField;
+
+    @Inject
+    public DataCloudLogInWindow(final DataCleanerConfiguration configuration,
+                                final UserPreferences userPreferences, WindowContext windowContext, AbstractWindow owner) {
+        super(windowContext, ImageManager.get().getImage("images/window/banner-logo.png"), owner);
+        _configuration = configuration;
+        _userPreferences = userPreferences;
+        _contentPanel = createContentPanel();
+        getRootPane().getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0), "EnterAction");
+        getRootPane().getActionMap().put("EnterAction", new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                signIn();
+            }
+        });
+
+    }
+
+    public static boolean mayIShowIt(UserPreferences userPreferences, DataCleanerConfiguration configuration) {
```

Comment:
I would rename to something like ```isRelevant(...)``` or something like that. In code there is no "I" :-)